### PR TITLE
Store sort order of groups in cookie

### PR DESCRIPTION
**Pull request by lucasvanlierop**

### CODE REVIEW

File: src/AppBundle/Controller/IndexController.php [hunk 1]
```diff
@@ -75,47 +76,41 @@ public function groupsAction(Request $request)
...
-        $sortColumn = $sort;
-        $sortDirection = 0;
-        if ($sort[0] === '-') {
-            $sortDirection = 1;
-            $sortColumn = substr($sort, 1);
-        }
-
-        $myGroups = $this->getMyGroups($type, $sortColumn, $sortDirection, $offset, $limit);
-
+        $myGroupsSortOrder = $this->createSortOrder($cookies, 'my_groups', $signedSort);
```
Comment (rjd22) [inline src/AppBundle/Controller/IndexController.php]: I would get the sortorder from the cookie here.

File: src/AppBundle/Controller/IndexController.php [hunk 2]
```diff
@@ -217,4 +211,26 @@ private function parsePanelsCookie(ParameterBag $cookies)
...
+     * @return SortOrder
+     */
+    private function createSortOrder(ParameterBag $cookies, $groupName, $defaultSignedName)
+    {
+        $signedSortFromCookie = json_decode($cookies->get(sprintf('group_%s_sort_order', $groupName)));
+
+        if ($signedSortFromCookie) {
+            try {
+                return SortOrder::createFromSignedName($signedSortFromCookie);
+            } catch (\Exception $ex) {
```
Comment (rjd22) [inline src/AppBundle/Controller/IndexController.php]: Import this one please :)

File: src/AppBundle/Controller/IndexController.php [hunk 3]
```diff
@@ -217,4 +211,26 @@ private function parsePanelsCookie(ParameterBag $cookies)
...
+                $this->get('logger')->warning($ex
+                    >getMessage());
```
Comment (rjd22) [inline src/AppBundle/Controller/IndexController.php]: I think this should be `$this->get('logger')->warning($ex->getMessage());`

File: tests/AppBundle/Model/SortOrderTest.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Tests\AppBundle\Model;
+
+use AppBundle\Model\SortOrder;
+use \InvalidArgumentException;
```
Comment (rjd22) [inline tests/AppBundle/Model/SortOrderTest.php]: You can remove this `\`

File: tests/AppBundle/Model/SortOrderTest.php [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+class SortOrderTest extends \PHPUnit_Framework_TestCase
```
Comment (rjd22) [inline tests/AppBundle/Model/SortOrderTest.php]: Import `PHPUnit_Framework_TestCase`

File: src/AppBundle/Model/SortOrder.php [hunk 6]
```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace AppBundle\Model;
+
+use InvalidArgumentException;
+
+final class SortOrder
+{
+    const DIRECTION_ASCENDING = 0;
```
Comment (rjd22) [inline src/AppBundle/Model/SortOrder.php]: `DIRECTION_ASCENDING = 'asc';`
`DIRECTION_DESCENDING = 'desc';`

File: src/AppBundle/Controller/IndexController.php (hunk 1 above)
Comment (lucasvanlierop) [inline src/AppBundle/Controller/IndexController.php]: Done

File: src/AppBundle/Controller/IndexController.php (hunk 2 above)
Comment (lucasvanlierop) [inline src/AppBundle/Controller/IndexController.php]: Done

File: src/AppBundle/Controller/IndexController.php (hunk 3 above)
Comment (lucasvanlierop) [inline src/AppBundle/Controller/IndexController.php]: Fixed

File: tests/AppBundle/Model/SortOrderTest.php (hunk 4 above)
Comment (lucasvanlierop) [inline tests/AppBundle/Model/SortOrderTest.php]: Done

File: tests/AppBundle/Model/SortOrderTest.php (hunk 5 above)
Comment (lucasvanlierop) [inline tests/AppBundle/Model/SortOrderTest.php]: Done

File: src/AppBundle/Model/SortOrder.php (hunk 6 above)
Comment (lucasvanlierop) [inline src/AppBundle/Model/SortOrder.php]: Fixed

**Comment by lucasvanlierop** [comment]
@Tjeerd ik merge 'm vast mocht je nog iets spotten hoor ik het graag!

**Pull request merged**